subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java

```diff
@@ -581,4 +582,39 @@
      */
     @Incubating
     Dependency testFixtures(Object notation, Action<? super Dependency> configureAction);
+
+    /**
+     * Allows fine tuning what variant to select for the target dependency. This can be used to
+     * specify a classifier, for example.
+     *
+     * @param dependencyProvider the dependency provider
+     * @param variantSpec the variant specification
+     * @return a new dependency provider targetting the configured variant
+     * @since 6.8
+     */
+    @Incubating
+    Provider<MinimalExternalModuleDependency> variantOf(Provider<MinimalExternalModuleDependency> dependencyProvider, Action<? super ExternalModuleDependencyVariantSpec> variantSpec);
+
+    /**
+     * Configures this dependency provider to select the platform variant of the target component
+     * @param dependencyProvider the dependency provider
+     * @return a new dependency provider targetting the platform variant of the component
+     * @since 6.8
+     */
+    @Incubating
+    default Provider<MinimalExternalModuleDependency> platform(Provider<MinimalExternalModuleDependency> dependencyProvider) {
```

Comment:
This signature means it will not work for the [`Provider` support](https://docs.gradle.org/6.5/release-notes.html#improvements-for-plugin-authors) introduced in 6.5.
While that may not be critical to have it right now, I would think twice before adding this if we cannot evolve it in a compatible way to have such support later.